src/pelicanfs/core.py

```diff
@@ -206,20 +251,30 @@ async def wrapper(self, *args, **kwargs):
             path = args[0]
             if isinstance(path, str):
                 parsedUrl = urllib.parse.urlparse(path)
-                if parsedUrl.scheme == "http":
-                    cacheUrl = path
+                if parsedUrl.scheme == "http" or parsedUrl.scheme == "https":
+                    dataUrl = path
                 else:
-                    cacheUrl = await self.get_working_cache(parsedUrl.path)
+                    if self.specifiedCache:
+                        dataUrl = self.specifiedCache + "/" + parsedUrl.path
```

Comment:
Please go through the whole request and use the specific URL joining routines as appropriate.